auth-table-based-service/src/main/java/io/stargate/auth/table/AuthzTableBasedService.java

```diff
@@ -40,6 +47,22 @@ public ResultSet authorizedDataRead(
     // Cannot perform authorization with a table based token so just return
     return action.call();
   }
+  /**
+   * Authorization for data access is not provided by table based tokens so all authorization will
+   * be deferred to the underlying permissions assigned to the role the token maps to.
+   *
+   * <p>{@inheritdoc}
+   */
+  @Override
+  public CompletionStage<ResultSet> authorizedAsyncDataRead(
+      Supplier<CompletionStage<ResultSet>> action,
+      String token,
+      String keyspace,
+      String table,
+      List<TypedKeyValue> typedKeyValues,
+      SourceAPI sourceAPI) {
+    return CompletableFuture.completedFuture(null).thenCompose(__ -> action.get());
```

Comment:
What do you think about using this pattern in `AuthzJwtService` too? My point that this would unify all exception handling under the `CompletionStage` framework. Currently we have possibility of some exceptions being thrown directly (in `AuthzJwtService`) and some being wrapped in the `CompletionStage`. Granted, some direct throwables are possible with his approach too, but they are mostly limited to severe linkage errors, OOM and stuff... execution exceptions would certainly go inside the `CompletionStage`. WDYT?

Reply:
I agree we should consistently returns exceptions as a failed future.

`CompletionStage` transformations do so for any type of error:

> if a stage's computation terminates abruptly with an (unchecked) exception or error, then all dependent stages requiring its completion complete exceptionally as well, with a CompletionException holding the exception as its cause.

I think we can achieve the same in `AuthzJwtService` by just catching `Throwable` instead of `Exception` in the synchronous part.

Reply:
> I think we can achieve the same in AuthzJwtService by just catching Throwable instead of Exception in the synchronous part.

Yes, but it adds complexity to our code... just my 2 cents :)